plugins/modules/zypper_repository.py

```diff
@@ -32,13 +33,24 @@
         type: str
     repo:
         description:
-            - URI of the repository or .repo file. Required when state=present.
+            - URI of the repository or .repo file. Required when I(state=present) and I(action=add).
         type: str
-    state:
+    action:
```

Comment:
Please note that module options such as `action` are not a good idea (https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#designing-module-interfaces), in particular if `state` is already present.

All your actions should already be doable with `state`. Modifying should be done with `state=present` and passing other parameters.

Reply:
I hear you, but `state` with `[ "absent", "present" ]` is only relevant when using `add` or `remove`, but it's irrelevant when using `action` with `[ "modify" ]`. Introducing `action` was IMHO the only possibility to add `feature: modify`.
If you have a better idea, I am open to hear ...

Reply:
No, `state=present` usually also means modify (except in certain situations, like it hasn't been implemented yet).

Please remove the `action` option and change the code so that with `state=present` it creates (if it doesn't exist yet) or modifies (if it does exist and is changed).